pkg/utils/utils.go

```diff
@@ -53,12 +54,23 @@ func RetrieveVcConfigSecret(params map[string]interface{}, logger logrus.FieldLo
 
 	ns := "kube-system"
 	secretApis := clientset.CoreV1().Secrets(ns)
-	vsphere_secret := "vsphere-config-secret"
-	secret, err := secretApis.Get(vsphere_secret, metav1.GetOptions{})
+	vsphere_secrets := []string{"vsphere-config-secret", "csi-vsphere-config"}
+	var secret *k8sv1.Secret
+	for _, vsphere_secret := range vsphere_secrets {
+		secret, err = secretApis.Get(vsphere_secret, metav1.GetOptions{})
+		if err == nil {
+			logger.Infof("Retrieved k8s secret, %s", vsphere_secret)
+			break
+		}
+		logger.WithError(err).Infof("Failed to get k8s secret, %s", vsphere_secret)
+	}
+
+	// No valid secret found.
 	if err != nil {
```

Comment:
I may be wrong, but isn't the "err" actually out of scope by the time it reaches here? I believe the err here points to line 49

Reply:
err gets initialized in line 44.

Reply:
every time we use ":=" in go, that means the variable, e.g. `err` here, gets the whole scope of the method. At the line 60, "=" was used to update `err`. So, it was only update the value without changing the scope of variable.